mesh/mesh.go

```diff
@@ -487,6 +499,32 @@ func (msh *Mesh) getRunningLayerHash(layerID types.LayerID) (types.Hash32, error
 	return hash, nil
 }
 
+// GetLayerHash returns layer hash for received blocks
+func (msh *Mesh) GetLayerHash(layerID types.LayerID) types.Hash32 {
+	h := types.Hash32{}
+	bts, err := msh.general.Get(msh.getLayerHashKey(layerID))
+	if err != nil {
+		return types.Hash32{}
+	}
+	h.SetBytes(bts)
+	return h
+}
+
+// GetLayerHashBlocks returns blocks for given hash
+func (msh *Mesh) GetLayerHashBlocks(h types.Hash32) []types.BlockID {
+	layerIDBytes, err := msh.general.Get(h.Bytes())
+	if err != nil {
+		msh.Warning("requested unknown layer hash %v", h.Hex())
```

Comment:
why not return error? otherwise you won't be able to know empty layer and not existing layer

Reply:
opened #2347